### ISSUE TITLE
Tools in progress

### ISSUE THREAD

**Issue by mblue9** [galaxyproject/tools-iuc#1657]
Hello all,

Just wanted to let people know what tools I'm working on (or planning to work on in the near future), in case anyone else is interested in these. 
I'm concentrating on RNA-seq/ChIP-seq/ATAC-seq workflows at the moment and will get onto these things as I can get the time.

**EGSEA** for Gene Set Testing: https://github.com/mblue9/tools-iuc/tree/egsea/tools/egsea
- currently adding in option to take in the single counts files from featurecounts (similar to deseq2/edger)
- first version will use egsea.cnt, later will add egsea.ma, egsea.ora etc

**fgsea** for Gene Set Testing: https://github.com/mblue9/tools-iuc/tree/fgsea/tools/fgsea
- currently modifying Mark Dunning's wrapper to e.g. use a loc file for the Broad gmt files

**MAGeCK** for CRISPR: https://github.com/mblue9/tools-iuc/tree/mageck/tools/mageck
- need to add more tests
- tidy up help
- see why mageck test pdf works but not mageck count

**Glimma** for interactive visualisation of RNA-seq 
- I haven't started on this yet but I plan to add it in

**CNVkit**
- also haven't started on this but I was thinking it would be good to try to update the existing old Galaxy wrapper here: https://github.com/etal/cnvkit/blob/master/galaxy/cnvkit_tool.xml

I've made a start on some other wrappers too:
- creating wrappers for **JAFFA** and **EricScript** for fusions, but discovered I need to make small reference genome files for testing as they both use ~8GB of custom reference files for human, when I get a chance I'll go back to try to get those working.
- updating the **Venn Diagram** tool to use rpy2
- updating **DiffBind**
- seeing if I can add in an option to use the inbuilt annotation to **featureCounts**

**Comment by mvdbeek** [comment]
That looks like a great list! I have successfully used https://cran.r-project.org/web/packages/VennDiagram/index.html for some one-off plotting, the advantage is that the package can draw the sizes to scale and deal with multiple datastes (5 is the max, I think) ... even if in this case one should probably resort to bar charts or some other plotting (https://github.com/hms-dbmi/UpSetR ?).

**Comment by mblue9** [comment]
Thanks @mvdbeek that's good to know! I had started on updating this Venn diagram tool from the toolshed https://github.com/peterjc/pico_galaxy/tree/master/tools/venn_list but I'll take a look at that one too. As a Venn diagram tool would be useful for lots of things I think so it would be good to have an IUC-one. Although I don't know yet when I'll find the time to get back onto Venns.

**Comment by mblue9** [comment]
Priorities have changed since I wrote this so I've changed this to a "might get worked on" and removed the tools that @mvdbeek has added (RUVseq and UpSetR), thanks for adding those!

**Comment by mvdbeek** [comment]
I do have to do some volcano plots and ma-plots using labels and group highlights -- I might look into that very soon if you haven't already.

**Comment by bebatut** [comment]
@mvdbeek did you add UpSetR?

**Comment by mvdbeek** [comment]
Yes, as wrapped in [intervene](https://intervene.readthedocs.io/en/latest/index.html) -- https://github.com/galaxyproject/tools-iuc/pull/2074

**Comment by bebatut** [comment]
I totally missed it. Thanks!

**Comment by ic4f** [comment]
I'll take on the CVNkit wrapper, unless anyone is working on it.
